snapshots/src/unarchive.rs

```diff
@@ -52,14 +52,14 @@ pub fn streaming_unarchive_snapshot(
             (
                 decompressor,
                 file_creator(write_buf_size, &io_setup, move |file_info| {
-                    let result = file_sender.send(file_info.path);
+                    let result = file_sender.send(file_info);
                     if let Err(err) = result {
                         panic!(
                             "failed to send path '{}' from unpacker to rebuilder: {err}",
-                            err.0.display(),
+                            err.0.path.display(),
                         );
                     }
-                    Some(file_info.file)
+                    None
```

Comment:
We return `None` here now to tell the file creator to *not* close the file upon completion. Is that correct? If yes, can you add a comment here saying as much, please?

Reply:
correct, added comment